### PR TITLE
Add MSVC Spectre mitigation  

### PR DESCRIPTION
**Pull request by SergioRZMasson**
Added the /Qspectre MSVC compiler option when supported (Visual Studio 2017 version 15.5.5 and all later versions). This should mitigate certain Spectre variant 1 security vulnerabilities.

More information:
https://learn.microsoft.com/en-us/cpp/build/reference/qspectre?view=msvc-170

### CODE REVIEW

**Comment by SergioRZMasson** [comment]
@walbourn , is there anything we need to do to move this forward?

**Comment by walbourn** [comment]
My general concern is that Spectre-mitgated libraries are *not* installed by default with the C++ Workload in Visual Studio, so these changes as they are will break for many people.

I think the better solution here is to do the following:

(A) For all users of the vcxproj scenario, they can easily add it externally via ``/p:SpectreMitigation=Spectre`` (by design). I'll make sure to add a note about this to my readmes/docs. I can also add a flavor of the libs to the NuGet to include this mitigation.

(B) For the CMakeLists.txt, keep the new option but have it guarded by a build option.

File: DirectXTex/DirectXTex_Windows10_2022.vcxproj [hunk 1]
```diff
@@ -107,41 +107,49 @@
     <ConfigurationType>StaticLibrary</ConfigurationType>
     <UseDebugLibraries>true</UseDebugLibraries>
     <PlatformToolset>v143</PlatformToolset>
+    <SpectreMitigation>Spectre</SpectreMitigation>
```
Comment (walbourn) [inline DirectXTex/DirectXTex_Windows10_2022.vcxproj:110]: Note that UWP does NOT support Spectre libs.

**Comment by SergioRZMasson** [comment]
BTW, we consume DirectXTex from the nuget packages. So my goal is to have the binaries in the package been built with /Qspectre flag.

**Comment by walbourn** [comment]
> BTW, we consume DirectXTex from the nuget packages. So my goal is to have the binaries in the package been built with /Qspectre flag.

Understood. I'm adding a second library to the NuGet package to support this. See https://github.com/microsoft/DirectXTex/issues/294

**Comment by walbourn** [comment]
CMake updates committed

https://github.com/microsoft/DirectXTex/commit/d91675bde6db83a577ffbf6f657815a333864b86

**Pull request closed without merging**